components/raftstore/src/store/snap.rs

```diff
@@ -1338,34 +1260,17 @@ where
             };
         }
 
-        let (dir, snap_size) = {
```

Comment:
I remember there was a deadlock problem.

Reply:
What deadlock problem? All places requires `rl` except `regsiter`, `deregister` and `init`. So it seems ok.